File: Java/Labs/Module 3/Lab 2/Q1 - Sliding Subarray Beauty/Main.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        String[] inputArr = scanner.nextLine().trim().split(" ");
        int[] arr = new int[inputArr.length];
        for (int i = 0; i < inputArr.length; i++) {
            arr[i] = Integer.parseInt(inputArr[i]);
        }
        
        int k = scanner.nextInt();
        int n = scanner.nextInt();
        
        List<Integer> result = slidingSubarrayBeauty(arr, k, n);
        for (int num : result) {
            System.out.print(num + " ");
        }
    }

    public static List<Integer> slidingSubarrayBeauty(int[] arr, int k, int n) {
        List<Integer> result = new ArrayList<>();

        for (int i = 0; i <= arr.length - k; i++) {
            int[] subarray = Arrays.copyOfRange(arr, i, i + k);
            Arrays.sort(subarray);
            result.add(subarray[n - 1]);
        }

        return result;
    }
}
                                                                                                                            

File: Python/Labs/Module 3/Lab 2/Q1 - Sliding Subarray Beauty/main.py
def gsb(arr,k,n):
    result=[]
    for i in range(len(arr)-k+1):
        subarray=arr[i:i+k]
        subarray.sort()
        result.append(subarray[n-1])
    return result

oldnums=input().split()
nums=[]
for i in oldnums:
    nums.append(int(i))
k=int(input())
x=int(input())
ans=gsb(nums,k,x)
for i in range(len(ans)-1):
    print(ans[i],end=" ")
print(ans[-1])
                                                                                                                            